pkg/parsing/impl/validator_test.go

```diff
@@ -116,7 +116,7 @@ func TestReadRunSQL(t *testing.T) {
 				if tc.expErrType == nil {
 					require.NoError(t, err)
 					require.NotNil(t, rs)
-					q, err := rs.GetQuery()
+					q, err := rs.GetQuery(nil)
```

Comment:
We can decide if we want to allow passing a `nil` resolver or not.
The reality, is that in the real validator implementation a non-nil value is always passed. Maybe for tests is useful.

So we have two options:
1. We allow `nil` and if `go-sqlparser` detects that the query has some `TXN_HASH()` but the resolver is `nil`, it can return an error like "resolver needed" or similar. But if that query doesn't have any custom function, it isn't a problem to have received a `nil` resolver because it wasn't needed.
2. We don't allow `nil` values, and we fail early at the `go-validator` level. For tests, we can create a dummy resolver that does nothing or panics.

I don't have a particularly strong opinion, since this is mostly for tests.

Reply:
1 sounds good!